android/src/main/kotlin/za/co/britehouse/flutter_microsoft_authentication/FlutterMicrosoftAuthenticationPlugin.kt

```diff
@@ -73,11 +96,11 @@ class FlutterMicrosoftAuthenticationPlugin : MethodCallHandler {
 
     @Throws(IOException::class)
     private fun getConfigFile(path: String): File {
-        val key: String = mRegistrar.lookupKeyForAsset(path)
-        val configFile = File(mainActivity.applicationContext.cacheDir, "config.json")
+        val key: String = binding!!.flutterAssets.getAssetFilePathByName(path)
```

Comment:
we should avoid forced unwrapping and handle it accordingly

Reply:
The method should only be called after plugin initializes, so the binding will not be null; therefore something should go very wrong for this to be nulled and not reassigned, hence the program should fail so we'd know to fix it

Reply:
but that's an issue as it is a third party framework where the behaviour could change, we should not rely on assumption, instead have safeguard or fail fast approach in place

Reply:
This is a new Flutter android plugin behavior that we are supporting on a 3rd party fork. The behavior doesn't change here unless we change it

Reply:
IMHO, I would avoid forced unwrapping wherever possible and I could see we could do as well here, I would still wary of plugin behaviour and instead prefer to have my own safe guard in place

Reply:
By default the binding is null, I'd suggest taking a safe approach and not force a non-null here..
You can add a check if binding is not null, so that the compiler smart casts the value.
Or write something like binding?.let{
<code here>
}

Reply:
Binding is `null` only before `onAttachedToEngine()`  and is `null` again only after `onDetachedFromEngine()`. All of the plugin methods, including `getConfigFile()` should execute in between the two aforementioned engine attachement calls; hence `binding` should never be `null` during `getConfigFile()`. 
If somehow it happens so that it would be `null`, then app should fail, because Method channels/Flutter Plugin do not work as expected and we don't really have a way to recover from it

Reply:
Updated implementation